programs/zstdcli.c

```diff
@@ -824,7 +823,7 @@ int main(int argCount, const char* argv[])
                         DISPLAYLEVEL(4, "[TRACE] call read function is finished\n");
                         DISPLAYLEVEL(4, "[TRACE] extendedFileNamesTable:\n");
 
-                        unsigned i = 0;
+                        unsigned i;
```

Comment:
Open a new block here, and close it right after the `for` loop.

Do the same for each `for` loop using an `i` counter.
There is no reason for a counter `i` to be shared between loop, because the outcome from a previous loop is not applicable to the next loop. They are effectively independent counters.